Readme + Additions

bucaran opened the pull request:
+ Add getopts plugin README.
+ Add fish-spec plugin README. (#328)
+ Update README with information about tests.
+ Update plugins README with information about new plugins: msg, getopts, fish-spec.

bpinto commented:
This looks terrific!!! Amazing job
I will review it the first time I am available tomorrow

bucaran commented:
:+1:

bucaran commented:
@bpinto Ping.

bpinto commented:
Ready now?

bucaran commented:
Sorry, I wasn't clear enough. Instead of WIP, I probably meant to say _there are still rooms for improvement_, but other than that, yes this is ready for merge. If you can spot any typos or anything else you want to add, let me know and I'll fix those, before rebasing and pushing again.

bpinto commented on plugins/fish-spec/README.markdown:
```diff
@@ -0,0 +1,87 @@
+Just like any other plugin in Oh-My-Fish, you can start using `fish-spec` in two ways:
+
+1. Adding the plugin to the `$fish_plugins` global variable declared in your fish configuration file, usually in  `~/.config/fish/config.fish`.
+2. Directly importing the library into your fish file via [import](https://github.com/bpinto/oh-my-fish/blob/master/functions/import.fish).
```
Maybe change the order? As the latter is what we suggest.

bucaran replied:
:+1:

bpinto commented on plugins/fish-spec/README.markdown:
```diff
@@ -0,0 +1,87 @@
+
+As of now, the most common use case is the second method, as you will probably want to test your own plugins or libraries.
+
+Inside your project's directory create a new `spec` folder (recommended) and add all your _spec_ files inside. _Spec_ files are regular fish files that must look like `*.spec.fish` and contain your tests.
```
I would remove the _recommended_, as it's kinda like mandatory.

bucaran replied:
:+1:

bpinto commented on plugins/fish-spec/README.markdown:
````diff
@@ -0,0 +1,87 @@
+
+```fish
+import plugins/fish-spec
+import path/to/the-library # plugins/the-library
````
Remove the comment and do `import plugins/library-name` directly?

bucaran replied:
:+1:

bpinto commented on plugins/fish-spec/README.markdown:
```diff
@@ -0,0 +1,87 @@
+
+Under the hood, _expect_ checks an _actual_ value, usually a relevant result from your test unit, is equal to, not equal to, etc., to an _expected_ value, as determined by your test. Below are the list of conditions available to use with `expect`:
+
+* __--to-equal__             `<actual>` value equals `<expected>` value
```
I think we could add more details to these methods, as for instance, we have `--to-contain-all --from`

bucaran replied:
Done!

bpinto commented:
Magnific!

bpinto commented on plugins/getopts/README.markdown:
````diff
@@ -0,0 +1,115 @@
+
+## Install
+
+```fish
````
We have this plugin builtin into oh-my-fish.

bucaran replied:
Of course :+1:

bpinto commented on plugins/getopts/README.markdown:
```diff
@@ -0,0 +1,115 @@
+
+## Tests
+
+Tests are implemented via [Oh-My-Fish](https://github.com/bpinto/oh-my-fish) spec runner _fish-spec_.
```
Not necessary?

bucaran replied:
:+1:

bpinto commented on plugins/getopts/README.markdown:
```diff
@@ -0,0 +1,115 @@
+
++ `getopts` will return the remaining operands when the end of the options is reached, i.e, a `--` argument that is not an option is found, or an argument that does not begin with `-` is found.
+
+## Environment Variables
```
I don't think this is necessary for people that want to learn how to use the tool.

bucaran replied:
Well, yes, but it does not hurt either.

bucaran commented:
Also went ahead and added a small _contributing_ section to the main readme (maybe we should add a CONTRIBUTING.md later on), rename `.markdown` to `.md` which is shorter, added links to each plugin in the plugins readme and removed some from deprecated plugins.

bpinto commented on README.md:
```diff
@@ -83,6 +85,12 @@ Substitute /bin/bash with /bin/tcsh or /bin/zsh as appropriate.
     rm -rf ~/.oh-my-fish
 
 
+## Contributing
```
We can remove that "Send us your theme".

bucaran replied:
What about plugins and other kind of contributions?

bucaran replied:
OK

## Contributing

Contributions are always welcome. Send us your theme / plugin. 

> __Note__: Theme authors, make sure to include a screenshot in your pull request. A nifty way is too post a comment with the image and link directly to it.

bpinto replied:
What did you miss on the readme the first time you contributed? :)

bucaran replied:
What do you mean? I don't remember, but the _note_ seems pretty spot on since people are always opening PRs without the image.

bpinto replied:
I need to see how github handles it, but I know there is a way to display a little readme whenever a person creates a pull request.

bucaran replied:
That would be cool.

bpinto commented on plugins/fish-spec/README.md:
````diff
@@ -0,0 +1,114 @@
+spec.run $argv
+```
+
+## API
````
Awesome!

bpinto commented on plugins/fish-spec/README.md:
````diff
@@ -0,0 +1,114 @@
+```
+
+## FAQ
+1. __How to use `fish-spec` without Oh-My-Fish?__
````
Too bad that PR to add fish-spec to fish-shell had no replies. =/

bucaran replied:
Well, `fish-spec` works and it rocks. At some point we should get around revising it and improving it, in fact let me open an issue on that.

bpinto replied:
Sure, we need to add more assertions, but apart from a couple I have in my mind, I don't have anything to add for the moment. :P

bpinto commented on plugins/getopts/README.md:
```diff
@@ -0,0 +1,110 @@
+
++ `getopts` will return the remaining operands when the end of the options is reached, i.e, a `--` argument that is not an option is found, or an argument that does not begin with `-` is found.
+
+## Environment Variables
```
You could say it makes no harm, and I would say the opposite, but if you want it. :+1: 
I like to keep it simple. And I don't think a `getopts` consumer will ever need to know about this.

bucaran replied:
Mmm, I got it. Let's remove it from here, but add a link in the plugins README to my `bucaran/getopts` for issues where the full documentation is available, at least this part.

bpinto replied:
:+1:

bpinto commented on plugins/README.md:
```diff
@@ -0,0 +1,41 @@
+# Plugins
+* [__android-sdk__](https://github.com/bpinto/oh-my-fish/tree/master/plugins/android-sdk) – [Android SDK](http://developer.android.com/sdk/index.html) integration.
```
Nice linking.

bpinto commented on README.md:
```diff
@@ -13,12 +13,11 @@
 Oh-my-fish is built for recent versions of the fish shell, namely "fish 2.0".
 
-    curl -L https://github.com/bpinto/oh-my-fish/raw/master/tools/install.fish | fish
+`curl -L https://github.com/bpinto/oh-my-fish/raw/master/tools/install.fish | fish`
```
I prefer the way the old one rendered, it is more visible and this is the most important part of the readme. :P

bucaran replied:
My bad. You are right, I didn't check my MD preview for this one.

bpinto commented:
Just saw this last message about the `CONTRIBUTING.md`, doing it now, won't be thorough, but we keep improving it.

bucaran commented:
:+1:

bpinto commented:
I'll push the contributing guide PR, you need to rebase anyway. :)

bucaran commented:
OK

bucaran commented:
Ready.

bucaran commented:
I think we are ready. I removed the CONTRIBUTING link.

bpinto commented on README.md:
```diff
@@ -27,9 +28,12 @@ If you want to install it manually, keep reading.
+As an example, to enable rails and git plugins add this line `set fish_plugins git rails`
 to your configuration file.
 
+## Tests
```
I read it again and again, and I think this is out of context. We are on the readme, where people want to know how to use the framework, if they want to know how to contribute, they should read the contributing guide instead.

bucaran replied:
What part?

bpinto replied:
I was talking about this test part instead.

bucaran replied:
Removed.

bucaran commented:
Ready to merge?

bpinto commented on plugins/getopts/README.md:
````diff
@@ -0,0 +1,110 @@
+## Usage
+
+```fish
+Add `getopts` to your `$fish_plugins` or import directly into your library via `import plugins/getopts`.
````
I'm not sure if we need to say to add it to $fish_plugins. This is already the "default" way to include a plugin, but this one in specific, is probably only going to be used by importing inside a plugin.

bucaran replied:
Got you. Reworded.

bpinto commented on plugins/fish-spec/README.md:
```diff
@@ -0,0 +1,114 @@
+Before you can use `fish-spec`, you need to install [Oh-My-Fish](https://github.com/bpinto/oh-my-fish).
+
+## Usage
+Like other plugins in Oh-My-Fish, you can start using `fish-spec` in two ways:
```
Perhaps the same applies here, only talk about adding `import plugins/fish-spec`, as the other is the default way.

bucaran replied:
Cleared.

bpinto commented:
Final comments, we will get it merged after this. :+1:

bpinto commented:
Awesome, rebase and merge?

I just pushed merged a PR, so we need a rebase.

bucaran commented:
:+1:

bpinto commented:
Do you want to hit merge once tests are done?

bucaran commented:
Green. I am hitting merge.

bpinto commented:
:+1:
